plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -213,46 +206,51 @@ public void testColumnComment()
     @Test
     public void testPredicatePushdown()
     {
+        // varchar inequality
+        assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name != 'ROMANIA' AND name != 'ALGERIA'"))
+                .matches("VALUES " +
+                        "(BIGINT '1', BIGINT '1', CAST('ARGENTINA' AS varchar(25))), " +
+                        "(BIGINT '1', BIGINT '2', CAST('BRAZIL' AS varchar(25))), " +
+                        "(BIGINT '1', BIGINT '3', CAST('CANADA' AS varchar(25))), " +
+                        "(BIGINT '4', BIGINT '4', CAST('EGYPT' AS varchar(25))), " +
+                        "(BIGINT '0', BIGINT '5', CAST('ETHIOPIA' AS varchar(25))), " +
+                        "(BIGINT '3', BIGINT '6', CAST('FRANCE' AS varchar(25))), " +
+                        "(BIGINT '3', BIGINT '7', CAST('GERMANY' AS varchar(25))), " +
+                        "(BIGINT '2', BIGINT '8', CAST('INDIA' AS varchar(25))), " +
+                        "(BIGINT '2', BIGINT '9', CAST('INDONESIA' AS varchar(25))), " +
+                        "(BIGINT '4', BIGINT '10', CAST('IRAN' AS varchar(25))), " +
+                        "(BIGINT '4', BIGINT '11', CAST('IRAQ' AS varchar(25))), " +
+                        "(BIGINT '2', BIGINT '12', CAST('JAPAN' AS varchar(25))), " +
+                        "(BIGINT '4', BIGINT '13', CAST('JORDAN' AS varchar(25))), " +
+                        "(BIGINT '0', BIGINT '14', CAST('KENYA' AS varchar(25))), " +
+                        "(BIGINT '0', BIGINT '15', CAST('MOROCCO' AS varchar(25))), " +
+                        "(BIGINT '0', BIGINT '16', CAST('MOZAMBIQUE' AS varchar(25))), " +
+                        "(BIGINT '1', BIGINT '17', CAST('PERU' AS varchar(25))), " +
+                        "(BIGINT '2', BIGINT '18', CAST('CHINA' AS varchar(25))), " +
+                        "(BIGINT '4', BIGINT '20', CAST('SAUDI ARABIA' AS varchar(25))), " +
+                        "(BIGINT '2', BIGINT '21', CAST('VIETNAM' AS varchar(25))), " +
+                        "(BIGINT '3', BIGINT '22', CAST('RUSSIA' AS varchar(25))), " +
+                        "(BIGINT '3', BIGINT '23', CAST('UNITED KINGDOM' AS varchar(25))), " +
+                        "(BIGINT '1', BIGINT '24', CAST('UNITED STATES' AS varchar(25)))")
+                .isFullyPushedDown();
```

Comment:
isFullyPushedDown verifies results as well so explicit `matches` can probably be removed.

Reply:
Fixed